qirlib/CMakeLists.txt

```
@@ -0,0 +1,84 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+cmake_minimum_required(VERSION 3.10.0)
+
+project(qirlib_external)
+include(ExternalProject)
+
+# Environment variables
+# Download vars
+# - QIRLIB_DOWNLOAD_LLVM => If set, download pre-built LLVM binaries.
+# - QIRLIB_LLVM_BUILDS_URL => URL root where downloads are hosted.
+# - QIRLIB_LLVM_PKG_NAME => name of package to be downloaded/created
+# - CPACK_PACKAGE_FILE_NAME
+# Build vars
+# - QIRLIB_LLVM_TAG
+# Both
+# - CMAKE_INSTALL_PREFIX
+
+if (NOT $ENV{QIRLIB_DOWNLOAD_LLVM} MATCHES "false")
+  message (STATUS "Install LLVM to ${CMAKE_INSTALL_PREFIX}")
+
+  set(SHA256_NAME $ENV{QIRLIB_LLVM_PKG_NAME}.sha256)
+  message (STATUS "Download file: $ENV{QIRLIB_LLVM_BUILDS_URL}/$ENV{QIRLIB_LLVM_PKG_NAME}")
+  message (STATUS "Download sha:$ENV{QIRLIB_LLVM_BUILDS_URL}/${SHA256_NAME}")
+
+  # download the expected archive hash
+  file(DOWNLOAD "$ENV{QIRLIB_LLVM_BUILDS_URL}/${SHA256_NAME}" ${CMAKE_BINARY_DIR}/${SHA256_NAME} STATUS SHA256_DOWNLOAD_STATUS)
+  list(GET SHA256_DOWNLOAD_STATUS 0 SHA256_DOWNLOAD_STATUS)
+  if (NOT (${SHA256_DOWNLOAD_STATUS} EQUAL 0))
+    list(GET SHA256_DOWNLOAD_STATUS 1 ERROR_MESSAGE)
+    message(FATAL_ERROR "Failed to download sha256 hash: ${ERROR_MESSAGE}")
+  endif ()
+
+  # load the expected archive hash into a variable for cmake to check
+  file(STRINGS ${CMAKE_BINARY_DIR}/${SHA256_NAME} LLVM_SHA256_SUM REGEX [0-9a-f]+)
```

Can the regex contain capital letters? Underscores?
```suggestion
  file(STRINGS ${CMAKE_BINARY_DIR}/${SHA256_NAME} LLVM_SHA256_SUM REGEX [0-9a-fA-F]+)
```

---

@kuzminrobin Odd thing is they only contain capital letters.

---

Then the regex should also contain capitals only?
```suggestion
  file(STRINGS ${CMAKE_BINARY_DIR}/${SHA256_NAME} LLVM_SHA256_SUM REGEX [0-9A-F]+)
```